software/py/generate_stats.py

```diff
@@ -0,0 +1,227 @@
+#
+#   generate_stats.py
+#
+#   vanilla core stats extractor
+# 
+#   input: vanilla_stats.log
+#   output: execution_stats.log
+#
+#   @author Borna
+#
+#   How to use:
+#   python3 generate_stats.py {manycore_dim_y} {manycore_dim_x} {vanilla_stats.log}
+#
+#   ex) python3 generate_stats.py 4 4 vanilla_stats.log
+#
+#   {manycore_dim_y}  Mesh Y dimension of manycore
+#   {manycore_dim_x}  Mesh X dimension of manycore
+
+
+
+import sys
+import os
+import re
+from enum import Enum
+
+
+instructions_list = ['instr','fadd','fsub','fmul','fsgnj','fsgnjn',\
+                     'fsgnjx','fmin','fmax','fcvt_s_w','fcvt_s_wu',\
+                     'fmv_w_x','feq','flt','fle','fcvt_w_s','fcvt_wu_s',\
+                     'fclass','fmv_x_w','local_ld','local_st',\
+                     'remote_ld','remote_st','local_flw','local_fsw',\
+                     'remote_flw','remote_fsw','lr','lr_aq',\
+                     'swap_aq','swap_rl','beq','bne','blt','bge','bltu',\
+                     'bgeu','jalr','jal', 'sll',\
+                     'slli','srl','srli','sra','srai','add','addi','sub',\
+                     'lui','auipc','xor','xori','or','ori','and','andi',\
+                     'slt','slti','sltu','sltiu','mul','mulh','mulhsu',\
+                     'mulhu','div','divu','rem','remu','fence']
+
+miss_list = ['icache_miss', 'beq_miss', 'bne_miss', 'blt_miss',\
+                'bge_miss', 'bltu_miss', 'bgeu_miss', 'jalr_miss']
+
+stalls_list = ['stall_fp_remote_load','stall_fp_local_load',\
+               'stall_depend','stall_depend_remote_load',\
+               'stall_depend_local_load','stall_force_wb',\
+               'stall_ifetch_wait','stall_icache_store',\
+               'stall_lr_aq','stall_md,stall_remote_req','stall_local_flw']
+
+
+
+class Stats:
```

Comment:
I would call it something more descriptive like VanillaStatsParser.
the filename should match the name of the python class in the file.